### PR TITLE
ROX-14495: Fix failing admission controller test.

### PR DESCRIPTION
**Pull request by clickboo**
## Description

- Added code to turn on ROX_NEW_POLICY_CATEGORIES ff when postgres env is turned on.
- Added a UUID to the newPolicy name, to avoid, a duplicate policy name when being added (local runs failed with an error in post policy due to policy name collision)

## Checklist
- [x] Investigated and inspected CI test results
- [x] ~Unit test and regression tests added~
- [x] ~Evaluated and added CHANGELOG entry if required~
- [x] ~Determined and documented upgrade steps~
- [x] ~Documented user facing changes (create PR based on [openshift/openshift-docs]~(https://github.com/openshift/openshift-docs) and merge into [rhacs-docs](https://github.com/openshift/openshift-docs/tree/rhacs-docs))

If any of these don't apply, please comment below.

## Testing Performed
Fixed test hence automated CI run only

### CODE REVIEW

File: tests/admctrl_configmap_test.go [hunk 1]
```diff
@@ -20,6 +21,8 @@ import (
 )
 
 func TestAdmissionControllerConfigMap(t *testing.T) {
+	t.Setenv(features.NewPolicyCategories.Name(), "true")
```
Comment (md2119) [inline tests/admctrl_configmap_test.go:24]: Did you add this test through your mammoth feature PR? If not, why cannot we run this test anymore with feature flag off?

Comment (theencee) [reply tests/admctrl_configmap_test.go:24]: Wait both the postgres and non-postgres will run the same file. So does that mean the one that was working fine will now get the flag and thus might cause other issues?

Comment (clickboo) [reply tests/admctrl_configmap_test.go:24]: Let's wait for CI, but I don't think so.

Comment (md2119) [reply tests/admctrl_configmap_test.go:24]: Is the current fix enough on release builds e.g. nightly?

Comment (clickboo) [reply tests/admctrl_configmap_test.go:24]: Is there a way to find out until a nightly runs with it? Or is there a tag to run the nightlies on this PR?

Also looks like some infra blip causing the e2e tests to fail

Comment (clickboo) [reply tests/admctrl_configmap_test.go]: Alright, CI has other plans. So the wait and watch is longer. 🤷‍♀️

File: tests/admctrl_configmap_test.go [hunk 2]
```diff
@@ -51,10 +57,10 @@ func TestAdmissionControllerConfigMap(t *testing.T) {
 
 	policyServiceClient := v1.NewPolicyServiceClient(cc)
 	newPolicy := &storage.Policy{
-		Name:        "testpolicy_" + t.Name(),
+		Name:        "testpolicy_" + t.Name() + "_" + uuid.NewV4().String(),
 		Description: "test deploy time policy",
 		Rationale:   "test deploy time policy",
-		Categories:  []string{"test"},
+		Categories:  []string{"Test"},
```
Comment (theencee) [inline tests/admctrl_configmap_test.go:63]: Why does this make a difference? (or just a holdover from testing?)

File: tests/admctrl_configmap_test.go [hunk 3]
```diff
@@ -10,17 +10,23 @@ import (
...
+	"github.com/stackrox/rox/pkg/uuid"
 	"github.com/stretchr/testify/assert"
 	"github.com/stretchr/testify/require"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 )
 
 func TestAdmissionControllerConfigMap(t *testing.T) {
-	t.Skip("ROX-14495: This test is not working.")
+	if env.PostgresDatastoreEnabled.BooleanSetting() {
+		t.Setenv(features.NewPolicyCategories.Name(), "true")
```
Comment (theencee) [inline tests/admctrl_configmap_test.go:27]: so basically if this is a test for postgres, then this test will only run the policy category path. But is that accurate? Should it run whether or not policy categories are enabled?

Comment (clickboo) [reply tests/admctrl_configmap_test.go:27]: In postgres, new policy categories only, so yes. Am I missing anything?

Comment (clickboo) [reply tests/admctrl_configmap_test.go:27]: ( we plan to remove the new categories feature flag soon and keep postgres only - then it really won't matter)

**Comment by clickboo** [comment]
@charmik-redhat @md2119 
image

Seeing this failure, randomly at best.

**Pull request closed without merging**